python/mlx/optimizers.py

```diff
@@ -442,3 +442,58 @@ def apply_single(
         state["v"] = v
 
         return parameter - lr * m / (v + eps)
+
+
+class Lion(Optimizer):
+    r"""Implementation of the Lion optimizer [1]. 
+    Since updates are computed through the sign operation, they tend to 
```

Comment:
Since we are at it an empty line would be appreciated.

Also just saw that Sphinx would have picked up correctly just the 1st sentence so it wouldn't be too bad to have left it as it were initially but I still think it looks better with a single sentence summary first.